tools/gfx/render.h

```diff
@@ -1104,18 +1105,17 @@ class IRenderer: public ISlangUnknown
 
     struct Desc
     {
+        RendererType rendererType; // The underlying API/Platform of the renderer.
         int width = 0;                                  // Width in pixels
         int height = 0;                                 // height in pixels
         const char* adapter = nullptr;                  // Name to identify the adapter to use
         int requiredFeatureCount = 0;                   // Number of required features.
         const char** requiredFeatures = nullptr;        // Array of required feature names, whose size is `requiredFeatureCount`.
         int nvapiExtnSlot = -1;                         // The slot (typically UAV) used to identify NVAPI intrinsics. If >=0 NVAPI is required.
+        ISlangFileSystem* shaderCacheFileSystem = nullptr; // The file system for loading cached shader kernels.
```

Comment:
As discussed elsewhere, I like this idea, but it also has a lot of implications for how we do `.zip`-based shader caching if we want to support that. This is probably good to have as a stake in the ground, but shader caching needs a lot of top-down design work.